scripts/format_notebooks.py

```diff
@@ -181,6 +181,58 @@ def preprocess(self, nb, resources):
         return nb, resources
 
 
+class VersionCheckPreprocessor(preprocessors.Preprocessor):
+    metadata_tag = "VersionCheck"  # tag for added cells so that we can find them easily
+
+    try:
+        # determine the current stellargraph version by importing the library's version file
+        sys.path.append(
+            (
+                os.path.abspath(
+                    os.path.join(
+                        os.path.dirname(__file__), os.path.pardir, "stellargraph"
+                    )
+                )
+            )
+        )
+        import version as sgversion
+
+        stellargraph_version = sgversion.__version__
+    except ImportError as error:
+        print(f"ERROR: unable to import StellarGraph version: {error.message}")
+        stellargraph_version = None
+
+    version_check_code = f"""\
+# verify that we're using the correct version of StellarGraph for this notebook
+import stellargraph as sg
+from packaging import version
+
+if version.parse(sg.__version__) < version.parse("{stellargraph_version}"):
+    raise ValueError(
+        f"The installed version {{sg.__version__}} of StellarGraph is older than the version {stellargraph_version} required for this notebook. Please update StellarGraph, eg: pip install --upgrade stellargraph"
```

Comment:
On the topic of trade-offs, I think this message could possibly provide a bit more guidance: several instances of this problem seem to be users looking at develop notebooks, but using a recent release, meaning `pip install --upgrade stellargraph` may not actually fix the problem. Potentially we could do something like:

```python
if ...:
    raise ValueError(
        "This notebook requires StellarGraph version {stellargraph_version}, but found older version ({{sg.__version__}}) installed. Please check you're using notebooks from a matching git tag, or consider updating StellarGraph (for instance, 'pip install --upgrade stellargraph')"
    )
```

However, jamming everything into a single exception message is kinda annoying; I tried to rephrase the message above to make it shorter and thus make space for the extra explanation, but there's still more we could say (e.g. explaining how to get the git tag on the command line vs. GitHub, and detecting whether the required version is a prerelease (meaning `pip install` won't work) or a realised one (meaning it will)). The final formatting is also a bit suboptimal.

One way to handle this would be to do something like linking to an external document, where we can write as much as we want (and with better formatting). For instance, we could edit https://github.com/stellargraph/stellargraph/issues/1172 or add comments to it.

```python
if version.parse(sg.__version__) != version.parse("{stellargraph_version}"):
    raise ValueError(
        "This notebook requires StellarGraph version {stellargraph_version}, but found different version ({{sg.__version__}}) installed. Please see <https://github.com/stellargraph/stellargraph/issues/1172>."
    )
```

(I changed the `<` to ` !=` because the explanation on #1172 could talk about that case too.)

Another way would be to have a function like `sg.utils.check_version` that we call In each notebook like `sg.utils.check_version("{stellargraph_version}")`. Of course, older versions of `StellarGraph` won't have this function, so a user will get `AttributeError` instead of the useful message, but maybe we can manage that?

```python
import stellargraph as sg
try:
    sg.utils.check_version("{stellargraph_version}")
except AttributeError:
    raise ValueError("This notebook requires StellarGraph version {stellargraph_version}, but an older version is installed.  Please see <https://github.com/stellargraph/stellargraph/issues/1172>.")
```